### PR TITLE
Use `LiveListSrv` to reduce angular watchers

### PR DESCRIPTION
**Pull request by alxndrsn**
Issue: #1862

@garethbowen please review and merge if happy.

### CODE REVIEW

File: static/js/controllers/inbox.js [hunk 1]
```diff
@@ -37,6 +35,9 @@ require('moment/locales');
         }
       );
 
+      LiveListConfig();
+      CheckDate($scope);
```
Comment (garethbowen) [inline static/js/controllers/inbox.js]: This is part of a different commit - might need to merge from develop?

File: static/js/services/live-list.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
+// TODO cache activeDom; set it when loading a controller!!
+
+// medic-webapp specific config for LiveList.
+// This service should be invoked once at startup.
+angular.module('inboxServices').factory('LiveListConfig', [
+  '$parse', '$templateCache', 'LiveList',
+  function($parse, $templateCache, LiveList) {
+    // Configure LiveList service
+    return function() {
+      var contactTypes = [ 'district_hospital', 'health_center', 'clinic', 'person' ];
```
Comment (garethbowen) [inline static/js/services/live-list.js]: We copy and paste this everywhere - maybe add an angular constant for the array and then inject it everywhere instead?

File: templates/partials/contacts_list_item.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<!--
+  TODO li should hve optional class selected, calculated as follows:
+    {{contact._id === loadingContent || !loadingContent && contact._id === selected.doc._id ? 'selected' : ''}}
+  This highlight is not useful on mobile, and needs a click listener or similar implemented for desktop users.
+-->
```
Comment (garethbowen) [inline templates/partials/contacts_list_item.html]: Does this comment get stripped out?

File: tests/karma/unit/controllers/reports.js [hunk 4]
```diff
@@ -84,48 +91,43 @@ describe('ReportsCtrl controller', function() {
     chai.expect(scope.filterModel.type).to.equal('reports');
   });
 
-  it('updated reports when changed', function() {
+  it('updates reports when changed', function(done) {
+    DB = {
```
Comment (garethbowen) [inline tests/karma/unit/controllers/reports.js]: It'd be tidier to use the `KarmaUtils.mockDB` function here.

File: tests/karma/unit/controllers/tasks.js [hunk 5]
```diff
@@ -71,7 +81,8 @@ describe('TasksCtrl controller', function() {
       TaskGenerator.args[0][1](null, expected);
       chai.expect(TaskGenerator.callCount).to.equal(1);
       chai.expect(scope.filterModel.type).to.equal('tasks');
-      chai.expect(scope.tasks).to.deep.equal(expected);
+      chai.assert.isTrue(LiveList.tasks.update.calledWith(expected[0]));
+      chai.assert.isTrue(LiveList.tasks.update.calledWith(expected[1]));
```
Comment (garethbowen) [inline tests/karma/unit/controllers/tasks.js]: It'd be nice to stay consistent with the surrounding code and use expect or assert form but not both...

File: static/js/controllers/inbox.js (hunk 1 above)
Comment (alxndrsn) [inline static/js/controllers/inbox.js]: No, just moved the invokation.

File: templates/partials/contacts_list_item.html (hunk 3 above)
Comment (alxndrsn) [inline templates/partials/contacts_list_item.html]: Comment not stripped...yet: https://github.com/medic/medic-webapp/pull/1884

**Comment by alxndrsn** [comment]
Merged manually.

**Pull request closed without merging**